### ISSUE TITLE
[Masthead (All components under Masthead)]: QA: In 200% and 400% zoom, after opening the hamburger menu, expand 'Consulting & Services' element and user move to the next level of options.  Invoke 'Back' button and focus is going to the first interactive element but not to the triggering element.

### ISSUE THREAD

**Issue by Gopi916** [carbon-design-system/carbon-for-ibm-dotcom#8846]
### Description

Page Element:
Product & Solutions
Consulting & Services
Learn & Support
Explore More

Issue:
In 200% and 400% zoom, navigation menu is wrapped inside the hamburger. After opening the hamburger menu, expand 'Consulting & Services' element and user move to the next level of options.
Invoke 'Back' button and focus is going to the first interactive element but not to the triggering element.

Expected:
In 200% and 400% zoom, navigation menu is wrapped inside the hamburger. After opening the hamburger menu, expand 'Consulting & Services' element and user move to the next level of options.
Invoke 'Back' button and focus should go to the triggering element 'Consulting & Services'.

### Component(s) impacted

Masthead:
Default
With custom type ahead
Search open on load
With L1
With alternate logo and tool tip

### Browser

Firefox

### Carbon for IBM.com version

v1.34.0-rc.0

### Severity

Severity 4 = The problem is not visible to or noticeable to an average user. Affects minor functionality, no workaround needed.

### Application/website

Carbon for IBM.com

### Package

@carbon/ibmdotcom-web-components

### CodeSandbox example

https://carbon-design-system.github.io/carbon-for-ibm-dotcom/next/web-components/?path=/story/components-masthead--default

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon-for-ibm-dotcom/blob/main/.github/CODE_OF_CONDUCT.md)
- [ ] I checked the [current issues](https://github.com/carbon-design-system/carbon-for-ibm-dotcom/issues) for duplicate issues
